src/Meziantou.Analyzer/Rules/FileNameMustMatchTypeNameAnalyzer.cs

```diff
@@ -154,4 +154,21 @@ private static bool IsWildcardMatch(string input, string pattern)
         var wildcardPattern = $"^{Regex.Escape(pattern).Replace("\\*", ".*", StringComparison.Ordinal)}$";
         return Regex.IsMatch(input, wildcardPattern, RegexOptions.CultureInvariant | RegexOptions.IgnoreCase, TimeSpan.FromSeconds(1));
     }
+
+    private static string GetTypeKindDisplayString(INamedTypeSymbol symbol)
+    {
+        return symbol.TypeKind switch
+        {
+            TypeKind.Class when symbol.IsRecord => "record",
```

Comment:
You'll need to add `#if CSHARP_N_OR_GREATER` (there are other example in the codebase) to support older versions of Roslyn.

Reply:
I prefer using Language version than Roslyn version when it's a language feature as it is self-documented. The code base is not 100% matching this rule because of history, but new code should.